lib/python/script/utils.py

```diff
@@ -467,3 +471,104 @@ def clock():
     if sys.version_info > (3,2):
         return time.perf_counter()
     return time.clock()
+
+
+def legalize_vector_name(name):
+    if not name:
+        raise ValueError("name cannot be empty")
+    if re.match("[^A-Za-z0-9]", name[0]):
```

Comment:
In many database systems you cannot create a table with a number as first character in a table name. This is why Vect_legal_filename() also declares it as illegal. So you should probably do that here as well.

Reply:
Starting with _ is the not the best option. I would probably either just throw an error message or use a random letter. An underscore as first character would not pass that test and if I try to create a vector map starting with "_" v.edit complains and throws an error...

Reply:
`legalize_vector_name()` has now a customizable fallback prefix with "x" as a default.